backend/src/prereq_loader.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  Schedule,
+  INEUAndPrereq,
+  INEUOrPrereq,
+  ScheduleYear,
+  ScheduleTerm,
+  ScheduleCourse,
+} from "../../frontend/src/models/types";
+import DataLoader from "dataloader";
+import request from "request-promise";
+
+/**
+ * Courses should have at least a classId and a subject, in order to query them.
+ * May have a termId.
+ */
+interface SimpleCourse {
+  termId?: number;
+  subject: string;
+  classId: number;
+}
+
+/**
+ * The result of a prereq query for a class.
+ */
+interface NonEmptyQueryResult {
+  prereqs?: INEUOrPrereq | INEUAndPrereq;
+  coreqs?: INEUOrPrereq | INEUAndPrereq;
+}
+
+type PrereqQueryResult = undefined | NonEmptyQueryResult;
+
+// keep track of the total number of requests made.
+let totalNumRequests = 0;
+
+/**
+ * Asynchronously adds prereqs to a Schedule.
+ * Does not do mutation.
+ * @param schedules the schedule to add prereqs to
+ */
+export async function addPrereqsToSchedules(
+  schedules: Schedule[]
+): Promise<Schedule[]> {
+  // set the total to zero
+  totalNumRequests = 0;
```

Comment:
Seems like this was only so you could print the number of requests you ended up making, in which case you should probably just remove this